### PR TITLE
[l10n] Auto-mark strings for translation

### PR DESCRIPTION
**Pull request by himdel**
Fixes [AAH-800](https://issues.redhat.com/browse/AAH-800)

This PR:
* makes it possible to start marking strings for translation
  * this does not deal with plurals (`n_`) or non-translating (`N_`) yet, just the basic version
* auto-marks as many strings for translation as possible to do automatically (and fixes any errors generated during)

To achieve the first, this adds a simple `_` JS string template that either behaves as regular template string, or, when `localStorage.test_l10n = true` is set (+reload), marks every translatable string in french quotes...

```
_`Hello ${ 'world' }` #=> 'Hello world'
localStorage.test_l10n = true
_`Hello ${ 'world' }` #=> '>>Hello <<world'
```

l10n

The latter was just a series of perl invocations to convert anything that looks like a human string,
only strings starting with a capital letter could safely be converted in some cases,
multiline strings and strings with params are not quite possible to do with regex without some major preprocessing.

        perl -i -npe 's/(?<!_)`[A-Z][^`]*`/_$&/' src/**/*.*
        perl -i -npe s/=\''([A-Z][-a-zA-Z0-9_,. ]*)'\'/'={_`$1`}'/g src/**/*.*
        perl -i -npe s/\''([A-Z][-a-zA-Z0-9_,. ]+)'\''(?!:)/_`$1`'/g src/**/*.*
        perl -i -npe s/\''([A-Z][-a-zA-Z0-9_,.:?!@$%\[\]<> ]+)'\''(?!:)/_`$1`'/g src/**/*.*
        perl -i -npe s/=\''([A-Z][-a-zA-Z0-9_,.:?!@$%\[\]<> ]+)'\''/={_`$1`}'/g src/**/*.*
        perl -i -npe 's/"([A-Z][-a-zA-Z0-9_,.:?!@$%\[\]<> '\'']+)"(?!:)/_`$1`/g' src/**/*.*
        perl -i -npe 's/="([A-Z][-a-zA-Z0-9_,.:?!@$%\[\]<> '\'']+)"/={_`$1`}/g' src/**/*.*
        perl -i -npe 's/>([A-Z][-a-zA-Z0-9_,.:?!@$%\[\] '\'']+)</>{_`$1`}</g' src/**/*.*
        perl -i -npe 's/^(\s{4,})([A-Z][-a-zA-Z0-9_,.:?!@$%\[\] '\'']+)(\s*)$/$1\{_`$2`}$3/g unless /,\s*$/' src/**/*.*

---

(Adding lingui itself in a separate PR, there are still problems with macros and typescript.) ([AAH-803](https://issues.redhat.com/browse/AAH-803))
(Guidelines re pluralization, params coming also separately ([AAH-802](https://issues.redhat.com/browse/AAH-802))

Review: please tell me about any string that's marked incorrectly or shouldn't be marked at all
(If you see a string that should be marked and isn't, just fix it outside of this PR, this should add the bulk of the changes, so we can then finish the rest manually.)

### CODE REVIEW

File: src/api/active-user.ts [hunk 1]
```diff
@@ -45,7 +45,7 @@ class API extends HubAPI {
     if (DEPLOYMENT_MODE === Constants.INSIGHTS_DEPLOYMENT_MODE) {
       return new Promise((resolve, reject) => {
         reject(
-          'Use window.chrome.insights.auth to get tokens for insights deployments',
+          _`Use window.chrome.insights.auth to get tokens for insights deployments`,
```
Comment (ZitaNemeckova) [inline src/api/active-user.ts:48]: This is probably not meant to be translated.

File: src/api/base.ts [hunk 2]
```diff
@@ -76,7 +76,7 @@ export class BaseAPI {
       await (window as any).insights.chrome.auth.getUser();
     }
     if (DEPLOYMENT_MODE === Constants.STANDALONE_DEPLOYMENT_MODE) {
-      request.headers['X-CSRFToken'] = Cookies.get('csrftoken');
+      request.headers[`X-CSRFToken`] = Cookies.get('csrftoken');
```
Comment (ZitaNemeckova) [inline src/api/base.ts:79]: This shouldn't be translated.

File: src/components/collection-detail/collection-info.tsx [hunk 3]
```diff
@@ -102,12 +102,12 @@ export class CollectionInfo extends React.Component<IProps> {
...
             <Split hasGutter={true}>
-              <SplitItem className='install-title'>Installation</SplitItem>
+              <SplitItem className='install-title'>{_`Installation`}</SplitItem>
               <SplitItem isFilled>
                 <ClipboardCopy isReadOnly>{installCommand}</ClipboardCopy>
                 <div>
-                  <b>Note:</b> Installing collections with ansible-galaxy is
-                  only supported in ansible 2.9+
+                  <b>{_`Note:`}</b> Installing collections with ansible-galaxy
+                  is only supported in ansible 2.9+
```
Comment (ZitaNemeckova) [inline src/components/collection-detail/collection-info.tsx:110]: Not a whole thing.

File: src/components/patternfly-wrappers/toolbar.tsx [hunk 4]
```diff
@@ -106,7 +106,7 @@ export class Toolbar extends React.Component<IProps, IState> {
   }
 
   private handleEnter(e) {
-    if (e.key === 'Enter') {
+    if (e.key === _`Enter`) {
```
Comment (ZitaNemeckova) [inline src/components/patternfly-wrappers/toolbar.tsx:109]: This shouldn't be translated.

File: src/components/patternfly-wrappers/write-only-field.tsx [hunk 5]
```diff
@@ -9,7 +9,7 @@ interface IProps {
   onClear: () => void;
 
   /**
-    Component to display when the user is allowed to update this field.
+    {_`Component to display when the user is allowed to update this field.`}
```
Comment (ZitaNemeckova) [inline src/components/patternfly-wrappers/write-only-field.tsx:12]: This doesn't have to be translated.

File: src/components/repositories/remote-form.tsx [hunk 6]
```diff
@@ -206,7 +208,7 @@ export class RemoteForm extends React.Component<IProps, IState> {
         {!disabledFields.includes('requirements_file') && (
           <FormGroup
             fieldId={'yaml'}
-            label={'YAML requirements'}
+            label={_`YAML requirements`}
             labelIcon={
               <HelperText
                 content={
```
Comment (ZitaNemeckova) [inline src/components/repositories/remote-form.tsx:214]: The following needs to be translated.

File: src/containers/execution-environment-detail/tag-manifest-modal.tsx [hunk 7]
```diff
@@ -219,7 +219,7 @@ export class TagManifestModal extends React.Component<IProps, IState> {
               variant='info'
               title={<>Waiting for {pendingTasks} task(s) to finish.</>}
```
Comment (ZitaNemeckova) [inline src/containers/execution-environment-detail/tag-manifest-modal.tsx:220]: Missing translation.

File: src/containers/execution-environment-detail/tag-manifest-modal.tsx [hunk 8]
```diff
@@ -219,7 +219,7 @@ export class TagManifestModal extends React.Component<IProps, IState> {
...
             >
-              It's safe to close this window. These tasks will finish in the
+              {_`It's safe to close this window. These tasks will finish in the`}
               background.
```
Comment (ZitaNemeckova) [inline src/containers/execution-environment-detail/tag-manifest-modal.tsx:223]: Missing translation.

File: src/containers/execution-environment-detail/tag-manifest-modal.tsx [hunk 9]
```diff
@@ -368,15 +368,14 @@ export class TagManifestModal extends React.Component<IProps, IState> {
       this.setState({
         verifyingTag: false,
         tagInFormError:
-          'A tag may contain lowercase and uppercase ASCII ' +
+          _`A tag may contain lowercase and uppercase ASCII ` +
           'alphabetic characters, digits, underscores, periods, and dashes. A tag must not ' +
```
Comment (ZitaNemeckova) [inline src/containers/execution-environment-detail/tag-manifest-modal.tsx:372]: Missing translation. Probably worth to add `+` to the magic as well.

File: src/containers/group-management/delete-group-modal.tsx [hunk 10]
```diff
@@ -19,7 +19,7 @@ export class DeleteGroupModal extends React.Component<IProps> {
       <DeleteModal
         cancelAction={cancelAction}
         deleteAction={deleteAction}
-        title='Delete group?'
+        title={_`Delete group?`}
       >
         <b>{name}</b> will be permanently deleted.
```
Comment (ZitaNemeckova) [inline src/containers/group-management/delete-group-modal.tsx:24]: Missing translation.

File: src/containers/group-management/delete-group-modal.tsx [hunk 11]
```diff
@@ -29,7 +29,7 @@ export class DeleteGroupModal extends React.Component<IProps> {
           )}
```
Comment (ZitaNemeckova) [inline src/containers/group-management/delete-group-modal.tsx:29]: Line above is missing a translation.

File: src/containers/group-management/delete-group-modal.tsx [hunk 12]
```diff
@@ -39,7 +39,7 @@ export class DeleteGroupModal extends React.Component<IProps> {
               </List>
             </>
           )}
-          {users && !count && <p>No users will be affected.</p>}
+          {users && !count && <p>{_`No users will be affected.`}</p>}
           {!users && (
             <p>
               Checking for affected users... <Spinner size='sm' />
```
Comment (ZitaNemeckova) [inline src/containers/group-management/delete-group-modal.tsx:45]: Missing transaltion.

File: src/containers/group-management/group-detail.tsx [hunk 13]
```diff
@@ -502,7 +502,7 @@ class GroupDetail extends React.Component<RouteComponentProps, IState> {
       <DeleteModal
         cancelAction={() => this.setState({ showUserRemoveModal: null })}
         deleteAction={() => this.deleteUser(user)}
-        title='Remove user from group?'
+        title={_`Remove user from group?`}
       >
         <b>{user.username}</b> will be removed from <b>{group.name}</b>.
```
Comment (ZitaNemeckova) [inline src/containers/group-management/group-detail.tsx:507]: Missing translation.

File: src/containers/namespace-detail/namespace-detail.tsx [hunk 14]
```diff
@@ -258,7 +258,7 @@ export class NamespaceDetail extends React.Component<IProps, IState> {
               <div>
                 <ClipboardCopy isReadOnly>{repositoryUrl}</ClipboardCopy>
                 <div>
-                  <b>Note:</b> Use this URL to configure ansible-galaxy to
+                  <b>{_`Note:`}</b> Use this URL to configure ansible-galaxy to
```
Comment (ZitaNemeckova) [inline src/containers/namespace-detail/namespace-detail.tsx:261]: Not whole string is translated.

File: src/containers/token/token-insights.tsx [hunk 15]
```diff
@@ -41,21 +41,23 @@ class TokenPage extends React.Component<RouteComponentProps, IState> {
...
       <React.Fragment>
-        <BaseHeader title='Connect to Hub'></BaseHeader>
+        <BaseHeader title={_`Connect to Hub`}></BaseHeader>
         <Main>
           <section className='body pf-c-content'>
-            <h2>Connect Private Automation Hub</h2>
+            <h2>{_`Connect Private Automation Hub`}</h2>
             <p>
-              Use the <Link to={Paths.repositories}>Repository Management</Link>{' '}
+              Use the{' '}
```
Comment (ZitaNemeckova) [inline src/containers/token/token-insights.tsx:49]: This will need more love as a whole :)

File: src/containers/token/token-standalone.tsx [hunk 16]
```diff
@@ -24,28 +24,31 @@ class TokenPage extends React.Component<RouteComponentProps, IState> {
 
     return (
       <React.Fragment>
-        <BaseHeader title='Token management'></BaseHeader>
+        <BaseHeader title={_`Token management`}></BaseHeader>
         <Main>
           <section className='body pf-c-content'>
-            <h2>API token</h2>
+            <h2>{_`API token`}</h2>
```
Comment (ZitaNemeckova) [inline src/containers/token/token-standalone.tsx:30]: This will need moore love as a whole :)

**Comment by ZitaNemeckova** [comment]
Missed ones:
https://github.com/ansible/ansible-hub-ui/pull/677/files#diff-8de429a7987b7dba9fa4dc8ef827a0e0921367b513021a43510cfd046ef52051R108
https://github.com/ansible/ansible-hub-ui/pull/677/files#diff-c3fdde670bdf4d238ac93ff344f0f449e6726f2ade0c311302d90a830221fad8R168
https://github.com/ansible/ansible-hub-ui/pull/677/files#diff-6408e75c2647e066747d87aec1d7b10fe828deb92c1d9bd12f5f9da36a55d850R478
https://github.com/ansible/ansible-hub-ui/pull/677/files#diff-6408e75c2647e066747d87aec1d7b10fe828deb92c1d9bd12f5f9da36a55d850L479
https://github.com/ansible/ansible-hub-ui/pull/677/files#diff-6408e75c2647e066747d87aec1d7b10fe828deb92c1d9bd12f5f9da36a55d850R509
https://github.com/ansible/ansible-hub-ui/pull/677/files#diff-6408e75c2647e066747d87aec1d7b10fe828deb92c1d9bd12f5f9da36a55d850L510
https://github.com/ansible/ansible-hub-ui/pull/677/files#diff-ab381d664c6428e26251e4d37689a4a59d48683c18a20db798b20ef550c14d70R228-R229
https://github.com/ansible/ansible-hub-ui/pull/677/files#diff-122aa0190172bd16cfe442b8d1d5cd247da0fa8e460ed13753a9d5627012e84aR167
https://github.com/ansible/ansible-hub-ui/pull/677/files#diff-122aa0190172bd16cfe442b8d1d5cd247da0fa8e460ed13753a9d5627012e84aR172
https://github.com/ansible/ansible-hub-ui/pull/677/files#diff-500c8a23434698f2b74338c3bc3c942ca51ab18cd6e4d9d44b97f8c8fea5a970R88
https://github.com/ansible/ansible-hub-ui/pull/677/files#diff-7b1ddacf90732379f57d698c43c0e275880f7761c5a73698314fc7c1e42d6303R239
https://github.com/ansible/ansible-hub-ui/pull/677/files#diff-7b1ddacf90732379f57d698c43c0e275880f7761c5a73698314fc7c1e42d6303R252
https://github.com/ansible/ansible-hub-ui/pull/677/files#diff-42d740c83268618af4cf6990fd65f91b766ddc30e0a814bb8c28e51ef75f2673R157
https://github.com/ansible/ansible-hub-ui/pull/677/files#diff-42d740c83268618af4cf6990fd65f91b766ddc30e0a814bb8c28e51ef75f2673R166
https://github.com/ansible/ansible-hub-ui/pull/677/files#diff-42d740c83268618af4cf6990fd65f91b766ddc30e0a814bb8c28e51ef75f2673R172
https://github.com/ansible/ansible-hub-ui/pull/677/files#diff-d8af2481d918b523fd4937edf793ccb79b57bf9046aa97d90bef861513e298dcR56
https://github.com/ansible/ansible-hub-ui/pull/677/files#diff-75e6eb690bf7aa2afba84db83be9923820fa04d1a08a40a6757f2c3b8d56b646R136
https://github.com/ansible/ansible-hub-ui/pull/677/files#diff-75e6eb690bf7aa2afba84db83be9923820fa04d1a08a40a6757f2c3b8d56b646R143


Notes:
`String ${value}` or `String` seems to be skipped.
`+` was breaking it.
`<br>` and `<p>` were breaking it. (having a space between `>` and string?)

**Comment by himdel** [comment]
@ZitaNemeckova I've added a commit dealing with `` `Strings` ``, fixed some of it in the last commit, and added a small commit the changes any `_ => null` into `() => null`, to prevent confusion with gettext.

Might be ready :)

**Comment by himdel** [comment]
Rebased, ready again :)

File: src/components/my-imports/import-list.tsx [hunk 17]
```diff
@@ -147,7 +147,7 @@ export class ImportList extends React.Component<IProps, IState> {
   }
 
   private handleEnter(e) {
-    if (e.key === 'Enter') {
+    if (e.key === _`Enter`) {
```
Comment (ZitaNemeckova) [inline src/components/my-imports/import-list.tsx:150]: This shouldn't be translated.

File: src/components/patternfly-wrappers/compound-filter.tsx [hunk 18]
```diff
@@ -170,7 +170,7 @@ export class CompoundFilter extends React.Component<IProps, IState> {
   }
 
   private handleEnter(e) {
-    if (e.key === 'Enter') {
+    if (e.key === _`Enter`) {
```
Comment (ZitaNemeckova) [inline src/components/patternfly-wrappers/compound-filter.tsx:173]: This shouldn't be translated.

File: src/containers/execution-environment-detail/tag-manifest-modal.tsx [hunk 19]
```diff
@@ -158,7 +158,7 @@ export class TagManifestModal extends React.Component<IProps, IState> {
                 onChange={val => this.setState({ tagInForm: val })}
                 isDisabled={!!tagToVerify || verifyingTag || isSaving}
                 onKeyUp={e => {
-                  if (e.key === 'Enter') {
+                  if (e.key === _`Enter`) {
```
Comment (ZitaNemeckova) [inline src/containers/execution-environment-detail/tag-manifest-modal.tsx:161]: This shouldn't be translated.

**Review by ZitaNemeckova — APPROVED** [review]
Except 3 `Enter`s LGTM 👍

**Pull request merged**